.github/workflows/pytest.yml

```diff
@@ -67,68 +70,38 @@ jobs:
-          # Needed because release/non-release downloads are different structure
-          cd amlg && if [ ! -f *.whl ]; then mv */*.whl ./; fi
-
-      - name: Download howso-validator-py
-        if: contains(inputs.upstream-details, 'howso-validator-py')
+      - name: Download dependencies
         env:
           GH_TOKEN: ${{ secrets.HOWSOAI_WORKFLOW_AUTOMATION_CLASSIC_TOKEN }}
         run: |
-          run_type=$(echo "${{ inputs.upstream-details }}" | jq '".howso-validator-py"."run-type"')
-          run_id=$(echo "${{ inputs.upstream-details }}" | jq '".howso-validator-py"."run-id"')
-          gh $run_type download -D hsv -R "howsoai/howso-validator-py" -p "howso_validator-*-py3-none-any*" "$run_id"
-          # Needed because release/non-release downloads are different structure
-          cd hsv && if [ ! -f *.whl ]; then mv */*.whl ./; fi
-
-      - name: Download howso-validator-enterprise-py
-        if: contains(inputs.upstream-details, 'howso-validator-enterprise-py')
-        env:
-          GH_TOKEN: ${{ secrets.HOWSOAI_WORKFLOW_AUTOMATION_CLASSIC_TOKEN }}
-        run: |
-          run_type=$(echo "${{ inputs.upstream-details }}" | jq '".howso-validator-enterprise-py"."run-type"')
-          run_id=$(echo "${{ inputs.upstream-details }}" | jq '".howso-validator-enterprise-py"."run-id"')
-          gh $run_type download -D hsve -R "howsoai/howso-validator-enterprise-py" -p "howso_validator_enterprise-*-py3-none-any*" "$run_id"
-          # Needed because release/non-release downloads are different structure
-          cd hsve && if [ ! -f *.whl ]; then mv */*.whl ./; fi
-
-      - name: Download howso-synthesizer-py
-        if: contains(inputs.upstream-details, 'howso-synthesizer-py')
-        env:
-          GH_TOKEN: ${{ secrets.HOWSOAI_WORKFLOW_AUTOMATION_CLASSIC_TOKEN }}
-        run: |
-          run_type=$(echo "${{ inputs.upstream-details }}" | jq '".howso-synthesizer-py"."run-type"')
-          run_id=$(echo "${{ inputs.upstream-details }}" | jq '".howso-synthesizer-py"."run-id"')
-          gh $run_type download -D synth -R "howsoai/howso-synthesizer-py" -p "howso_synthesizer-*-py3-none-any*" "$run_id"
-          # Needed because release/non-release downloads are different structure
-          cd synth && if [ ! -f *.whl ]; then mv */*.whl ./; fi
+          while read -r repo; do
+            echo "Getting details for $repo"
+            run_type=$(echo "${{ inputs.upstream-details }}" | jq --arg repo "$repo" '".$repo"."run-type"')
+            run_id=$(echo "${{ inputs.upstream-details }}" | jq --arg repo "$repo" '".$repo"."run-id"')
+            # Amalgam and howso-engine downloads need to be placed in their embedded location
+            if [[ "$repo" == "amalgam" ]]; then
+              gh $run_type download -D amalgam/lib/$plat/amd64 -R "howsoai/$repo" -p "*${{ inputs.amalgam-plat-arch }}*" "$run_id"
```

Comment:
How does this work for arm platforms? I know BDD and recipes don't run on arm but hse-py does. This might caught problems with that.

Reply:
Since amd64 is hardcoded here, seems like this would be a problem.